ui/apps/platform/src/utils/searchUtils.test.js

```diff
@@ -393,4 +394,18 @@ describe('searchUtils', () => {
             expect(searchValueAsArray(['cluster', 'namespace'])).toEqual(['cluster', 'namespace']);
         });
     });
+
+    describe('convertToExactMatch', () => {
+        it('returns a non-string value unmodified', () => {
+            expect(convertToExactMatch(undefined)).toEqual(undefined);
+            expect(convertToExactMatch(null)).toEqual(null);
+            expect(convertToExactMatch(42)).toEqual(42);
+            expect(convertToExactMatch(['42'])).toEqual(['42']);
+            expect(convertToExactMatch({ key: 'value' })).toEqual({ key: 'value' });
+        });
+
+        it('returns a string value wrapped in bespoke regex for exact match', () => {
+            expect(convertToExactMatch('cluster')).toEqual('r/^cluster$');
```

Comment:
I know that most objects in Kubernetes are required to follow RFC1123 or RFC1035 naming schemes, but just to make sure: Is there a possibility that strings we put into `convertToExactMatch` might contain `^` or `$`?

Reply:
Interesting thought Matthias. What would be the behavior in a testcase like: 

```
expect(convertToExactMatch('r/^cluster$'))...;
```

could we assume that this would be 

```
toEqual('r/^cluster$')
```
?

Reply:
I was thinking about the behaviour of this function if a Namespace/Deployment named something like `namespace$prod` or `nginx^2`. 
But yours is a valid concern as well.
Right now, I expect neither of our cases is likely, as Kubernetes doesn't allow for these kind of names (AFAIK), and our Frontend only ever offers valid Kubernetes Namespace/Deployment selections. 
Therefore I would be fine with merging the code as-is, but I wanted to raise that thought nonetheless.

That said, I would expect `convertToExactMatch('r/^cluster$')` to yield `'r/^cluster$'`, correct.
Which leaves the question: Should `convertToExactMatch('cluster$dev')` yield `cluster$dev` (not changing the string)? Should we even bother adding these cases?

Reply:
For the record: I am fine merging that as it is if handling the extended cases would be too much hassle now. We can add more tests (and input validations) in a follow-up.

Reply:
Good comments on edge cases. I can add comments on the expected input, but for this use-case, the namespace and deployment names come from dropdown controls which are only populated with existing namespace and deployment names. So, if `^ns$` is not a valid Kubernetes identifier, it couldn't be injected into the menu and thus this function, unless the user hacked the FE code.

FYI, JS string templates are not magic like in Perl

```
convertToExactMatch('r/^cluster$');
```

would literally return
```
'r/^r/^cluster$$'
```

As Thorin Oakenshield used to say, "There is nothing like looking,  if you want to find something", so I tried some of these NS names.

```
$ kc create ns ^sec22$
The Namespace "^sec22$" is invalid:
* metadata.name: Invalid value: "^sec22$": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
* metadata.labels: Invalid value: "^sec22$": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')
```

```
$ kc create ns sec$22
namespace/sec2 created
```
(note that kube seems to have treated the first "$2" as some kind of encoding and removed it)


I couldn't even sneak a trailing '$' in.
```
$ kc create ns sec22$
The Namespace "sec22$" is invalid:
* metadata.name: Invalid value: "sec22$": a lowercase RFC 1123 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character (e.g. 'my-name',  or '123-abc', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?')
* metadata.labels: Invalid value: "sec22$": a valid label must be an empty string or consist of alphanumeric characters, '-', '_' or '.', and must start and end with an alphanumeric character (e.g. 'MyValue',  or 'my_value',  or '12345', regex used for validation is '(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?')
```